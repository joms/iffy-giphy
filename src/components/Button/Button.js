import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ onClick, children, disabled, ...rest }) => {
    const click = e => {
        if (!disabled) {
            onClick(e);
        }
    };

    return (
        <button onClick={click} disabled={disabled} {...rest}>
            {children}
        </button>
    );
};

Button.defaultProps = {
    onClick: () => null,
    disabled: false,
};

Button.propTypes = {
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func,
    rest: PropTypes.object,
    disabled: PropTypes.bool,
};

export default Button;
