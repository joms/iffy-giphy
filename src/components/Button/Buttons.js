import React from 'react';
import PropTypes from 'prop-types';
import Button from '.';

const Buttons = ({ children }) => <div className="buttons">{children}</div>;

Buttons.propTypes = {
    children: PropTypes.node.isRequired,
};

export default Buttons;
