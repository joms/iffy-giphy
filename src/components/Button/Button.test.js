import React from 'react';
import { shallow } from 'enzyme';
import Button from '.';

describe('<Button>', () => {
    const wrapper = shallow(<Button>Hello, world</Button>);
    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders text', () => {
        expect(wrapper.text()).toEqual('Hello, world');
    });

    it('clicks', () => {
        const onClick = jest.fn();
        const clickWrapper = shallow(<Button onClick={onClick}>Hello, world</Button>);
        clickWrapper.simulate('click');
        expect(onClick).toHaveBeenCalled();
    });

    it('does not click when disabled', () => {
        const onClick = jest.fn();
        const clickWrapper = shallow(
            <Button disabled onClick={onClick}>
                Hello, world
            </Button>
        );
        clickWrapper.simulate('click');
        expect(onClick).not.toHaveBeenCalled();
    });
});
