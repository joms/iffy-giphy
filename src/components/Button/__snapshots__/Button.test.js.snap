// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> matches snapshot 1`] = `
<button
  disabled={false}
  onClick={[Function]}
>
  Hello, world
</button>
`;
